rain_api_core/general_util.py

```diff
@@ -96,35 +103,38 @@ def log_context(**context):
 
 
 def get_log():
-
     loglevel = os.getenv('LOGLEVEL', 'INFO')
     logtype = os.getenv('LOGTYPE', 'json')
     if logtype == 'flat':
-        log_fmt_str = "%(levelname)s: %(message)s (%(filename)s line " + \
-                      "%(lineno)d/%(build_vers)s/%(maturity)s) - " + \
-                      "RequestId: %(request_id)s; OriginRequestId: %(origin_request_id)s; user_id: %(user_id)s; route: %(route)s"
+        log_fmt_str = (
+            "%(levelname)s: %(message)s (%(filename)s line %(lineno)d/%(build_vers)s/%(maturity)s) - "
+            "RequestId: %(request_id)s; OriginRequestId: %(origin_request_id)s; user_id: %(user_id)s; route: %(route)s"
+        )
     else:
-        log_fmt_str = '{"level": "%(levelname)s",  ' + \
-                      '"RequestId": "%(request_id)s", ' + \
-                      '"OriginRequestId": "%(origin_request_id)s", ' + \
-                      '"message": %(message)s, ' + \
-                      '"maturity": "%(maturity)s", ' + \
-                      '"user_id": "%(user_id)s", ' + \
-                      '"route": "%(route)s", ' + \
-                      '"build": "%(build_vers)s", ' + \
-                      '"filename": "%(filename)s", ' + \
-                      '"lineno": %(lineno)d } '
+        log_fmt_str = (
+            '{"level": "%(levelname)s",  '
+            '"RequestId": "%(request_id)s", '
+            '"OriginRequestId": "%(origin_request_id)s", '
+            '"message": "%(message)s", '
```

Comment:
Add quotes around message, which is still prone to errors if the message itself contains quotes. Same applies to all other values as well.

Reply:
I'm tempted to look for a way to call `json.dumps('message')` to ensure message is always valid json, but that seems difficult

Reply:
Completely rewrote the logger to guarantee that it always outputs valid JSON